### PR TITLE
Create a documentation page for `torch.ao.quantization.QConfig`

### PR DESCRIPTION
**Pull request by vkuzo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66222 Quantization docs: add pages for Numeric Suite (Eager and FX)
* #66210 Quantization documentation: move backend section down
* #66201 Quantization docs: rewrite API reference to be more automated
* #66198 Quantization docs: consilidate all API references on a single page
* **#66129 Create a documentation page for `torch.ao.quantization.QConfig`**
* #66125 Create separate documentation pages for quantization observers and fake_quants
* #66122 Create a documentation page for FX graph mode quantization APIs

Summary:

Adds a documentation page for `torch.ao.quantization.QConfig`. It is useful
for this to have a separate page since it shared between Eager and FX graph
mode quantization.

Also, ensures that all important functions and module attributes in this
module have docstrings, so users can discover these without reading the
source code.

Test plan:

```
cd docs
make html
python -m http.server
// open webpage, inspect it, renders correctly
```

Differential Revision: [D31447614](https://our.internmc.facebook.com/intern/diff/D31447614)

### CODE REVIEW

File: torch/ao/quantization/qconfig.py [hunk 1]
```diff
@@ -225,6 +293,9 @@ def configure_constructor_to_put_obs_on_module_device(original_constructor):
 
 
 def qconfig_equals(q1: QConfigAny, q2: QConfigAny):
+    """
```
Comment (jerryzh168) [inline torch/ao/quantization/qconfig.py:296]: nit: should we define the rules about when qconfigs are equal here as well?

File: docs/source/torch.quantization.qconfig.rst [hunk 2]
```diff
@@ -0,0 +1,8 @@
+.. _torch_quantization_qconfig:
```
Comment (jerryzh168) [inline docs/source/torch.quantization.qconfig.rst:1]: should we rename this file to torch.ao.quantization.qconfig.rst?

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31447614).

**Review by jerryzh168 — APPROVED** [review]

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31447614).

**Pull request closed without merging**